interfaces/repo.go

```diff
@@ -539,9 +539,19 @@ func (r *Repository) ResolveDisconnect(plugSnapName, plugName, slotSnapName, slo
 	}
 }
 
+// SlotValidator can be implemented by Interfaces that need to validate the slot before the security is lifted.
+type SlotValidator interface {
+	BeforeConnectSlot(slot *ConnectedSlot) error
+}
+
+// SlotValidator can be implemented by Interfaces that need to validate the slot before the security is lifted.
+type PlugValidator interface {
+	BeforeConnectPlug(slot *ConnectedPlug) error
+}
+
 // Connect establishes a connection between a plug and a slot.
 // The plug and the slot must have the same interface.
-func (r *Repository) Connect(ref ConnRef) error {
+func (r *Repository) Connect(ref ConnRef, plugAttrs map[string]interface{}, slotAttrs map[string]interface{}, policyCheck func(*ConnectedPlug, *ConnectedSlot) error) (*Connection, error) {
```

Comment:
Not a problem, but it would be nice to pack the extra arguments into a structure, in current form it's quite unreadable

```
type ValidationData struct {
    PlugAttrs map[string]interface{}
    SlotAttrs map[string]interface{}
    PolicyChecker func(*ConnectedPlug, *ConnectedSlot
}
..

_, err = repo.Connect(interfaces.ConnRef{
		PlugRef: interfaces.PlugRef{Snap: info11.Name(), Name: ifname},
 		SlotRef: interfaces.SlotRef{Snap: core11.Name(), Name: ifname},
}, ValidationData{})
```